Add Code of Conduct (#16473)
PrestaShop/PrestaShop: CODE_OF_CONDUCT.md

# PrestaShop Code of Conduct

English version 

## Purpose 

> This Code of Conduct outlines how to behave to participate in the PrestaShop open source project (the “Project”) as well as prohibited behaviours and reporting procedure for damaging behaviours and/or situations. 

Reviewer: Behavior is [generally uncountable](http://www.macmillandictionaryblog.com/language-tip-of-the-week-behaviou).

```suggestion
This Code of Conduct outlines how to behave to participate in the PrestaShop open source project (the “Project”) as well as prohibited behaviour and reporting procedure for damaging behaviour and/or situations. 
```

The Code of Conduct aims to create a positive, inspiring and welcoming community, whether you are a beginner or an expert, asking questions or seeking to share your knowledge. 

## Scope of application

> This Code of Conduct applies to all people contributing to the Project, especially through online contributions, whether it is repositories, forum posts or on social media. 

Reviewer: ```suggestion
This Code of Conduct applies to all people contributing to the Project, especially through online contributions, whether it is in code repositories, forum posts or on social media. 
```

Author: it's not only about code, but also discussions in issues and pull requests, ...

> It also applies to public spaces, during PrestaShop events, aka the PSD, PS Connect, meetups or any other event during which an individual or a group of individuals represents the Project. 

Reviewer: ```suggestion
It also applies to public spaces, during PrestaShop events, aka the PrestaShop Day (PSD), PSConnect, meet-ups or any other event during which an individual or a group of individuals represents the Project. 
```

The Code of Conduct applies to any member of the PrestaShop community, whether it is PrestaShop employees, partners, ambassadors, moderators or any other members on the contributory sites and PrestaShop social media. 

> ## Expected behaviours 

Reviewer: ```suggestion
## Expected behaviour
```

> The Code of conduct aiming to create a positive and benevolent community, the following principles must be followed in order to have constructive and respectful exchanges : 

Reviewer 2: This sentence is a bit weird no? The structure doesn't feel English.

Proposition: 
_The Code of conduct ~aiming~ **aims** to create a positive and benevolent community, **thus** the following principles must be followed in order to have constructive and respectful exchanges_

or just 

_The following principles must be followed in order to have constructive and respectful exchanges_

Author: I will update that. thanks.

- If you want to help others, be patient and open. It may be difficult to learn, especially for new members, to participate in the Project. Be supportive while they learn.
- If you need help, make it easy for other members to help you. The community is made possible thanks to all its volunteers. Don’t worry if other members or moderators suggest changes, edit or rephrase your question: they try to make your question understandable and interesting to as many people as possible.
- Ensure to be respectful, courteous and polite. It is essential that community members show respect for each other, even when there are different opinions. The community welcomes any person, of various backgrounds and identities, interested in the Project and wishing to participate in its development.

> - Ensure to adopt a respectful, courteous and temperate language. If someone points out that your comments and/or your behaviour is making others uncomfortable, stop doing it. Sometimes, people unconsciously say things that negatively affect others. Even if it wasn’t your intent, apologize. When you share feedback, avoid sarcasm.. Be open to receiving constructive feedback. 

Reviewer: Small typo after "sarcasm".

```suggestion
- Ensure to adopt a respectful, courteous and temperate language. If someone points out that your comments and/or your behaviour is making others uncomfortable, stop doing it. Sometimes, people unconsciously say things that negatively affect others. Even if it wasn’t your intent, apologize. When you share feedback, avoid sarcasm. Be open to receiving constructive feedback. 
```

- Ensure that any interaction you have within the community remains lawful. Only share content that can be shared (ie. contents and materials that you have created or contents and materials on which the author allowed their reproduction and communication) and please do not ever share any sensitive information, content, etc. 
- Ensure that discussions you have with other members stay within the framework of the Project. Apart from the situation when a module can resolve a problem encountered in the use of the PrestaShop solution or event sponsorship, spam or advertising messages are not welcome. 

This list is not comprehensive. We expect the code to be followed in spirit as much as by the book. 

> ## Unacceptable behaviours

Reviewer: ```suggestion
## Unacceptable behaviour
```

> Certain behaviours are considered unacceptable and constitute a violation of this Code of Conduct, including:

Reviewer: ```suggestion
Certain kinds of behaviour are considered unacceptable and constitute a violation of this Code of Conduct, including:
```

- Violation of rights, in particular, the right to privacy, intellectual property rights (such as copyright, trademark, patent law etc.), trade secrets, personal injury, etc. 
- Rudeness and mockery, 
- Insults, discriminations or personal attacks. This includes any language likely to offend or alienate people based on race, gender identity or expression, language, sexual orientation, disability, nationality, age, profession, physical appearance, religion, minority background,
- Sexual or mental harassment, 
- Verbal or physical violence, threats, intimidation, bullying, etc. This includes subtle belittlement message or unwelcoming language. No matter the intent, such behaviour can have a negative impact on others. 

> This Code of Conduct is not intended to be exhaustive or to detail all forms of unacceptable behaviours. It is not designed to rank the behaviours or to consider that some behaviours do not deserve to be punished. 

Reviewer: ```suggestion
This Code of Conduct is not intended to be exhaustive or to detail all forms of unacceptable behaviours. It is not designed to rank behaviours or to consider that some kinds of behaviour do not deserve to be punished. 
```

If a member feels harassed or in an uncomfortable situation, they will be asked to report the situation according to the procedure described hereafter. 

> Every report will be examined. 

Reviewer: ```suggestion
All reports will be examined. 
```

## Reporting issues

If you experience or witness misconduct, you may: 

> - Flag it thanks to the “report content” button on the contributory sites; 

Reviewer: ```suggestion
- Flag it using the “report content” button on the contributory sites; 
```

> - Address it directly, and let the person know how this behaviour affects you or others;

Reviewer: ```suggestion
- Address it directly, and let the person know how that their behaviour affects you or others;
```

- If the situation is an online event, you may flag the content to the following address: [https://www.prestashop.com/en/abuse](https://www.prestashop.com/en/abuse);
- If the situation happens during a physical event, you may address it directly to a member of the organization or a security officer. 

> Your report must include the followings: 

Reviewer: ```suggestion
Your report must include the following: 
```

> - Your contact information, where you can be reached. Anonymous report cannot be treated; 

Reviewer: ```suggestion
- Your contact information, where you can be reached. Anonymous reports cannot be processed; 
```

> - Your name and the name of the person or the people involved;

Reviewer: ```suggestion
- Your name and the name of the person or persons involved;
```

Reviewer 2: name(s) ?

> - A description of what happened, and if you believe the situation is ongoing; don’t forget to provide the link of the webpage or the discussion; 

Reviewer: ```suggestion
- A detailed description of what happened; and if you believe the situation is ongoing, don’t forget to provide the link of the webpage or the discussion; 
```

- Any additional information that may be helpful.

> After filing a report, the most adapted person will handle the situation in order to check it. This person may ask you additional questions and will determine, with you, how to respond and handle the situation. 

Reviewer: ```suggestion
After the report has been filed, it will be forwarded to the appropriate person for verification and processing. This person may ask you additional questions and will determine, with you, how to respond and handle the situation. 
```

If the person harassing you is a moderator, they will be asked to recuse themselves. If the report is filed by a moderator, another moderator will handle the situation. 

## Sanction 

Each situation will be handled on a case-by-case basis. 

In the event of a non-compliance with the principles in the Code of Conduct, the warning procedure shall be initiated: 
- On the first occurrence, the moderator will send a public or private warning, and the content may be modified or deleted, 
- According to the situation, the account of the person may be temporarily suspended, 

> - For certain cases, such as extreme behaviours that affect the community, threats, acts of violence, a ban of the community may be applied along with a ban to participate in an event. 

Reviewer: ```suggestion
- For certain cases, such as extreme behaviour that affects the community, threats or acts of violence, a ban from the community may be applied along with a ban from participation in an event. 
```

> In case of participation in an event, a member of the organization will contact the alleged offender directly to explain the facts alleged and to inform him of the actions that may be taken against him. 

Reviewer: Use of gender neutral pronouns

```suggestion
In case of participation in an event, a member of the organization will contact the alleged offender directly to explain the alleged facts and to inform them of the actions that may be taken against them. 
```

> In the event of repetitive misconduct, the member of the organization reserves the right to exclude a participant. 

Reviewer: ```suggestion
In the event of repetitive misconduct, organization members reserve the right to exclude a participant. 
```

> If you have any doubts about how the situation has been handled, you can contact PrestaShop via this

Reviewer: > you can contact PrestaShop

Do you mean the PrestaShop Company? Or PrestaShop maintainers? Or the PrestaShop Company Staff?

Author: that's a branding issue and we can not solve it in this document. the link to the form makes it clear that it's "PrestaShop" in a global way.
then it will be sorted and sent to the best folks.

[contact form](https://www.prestashop.com/en/abuse). 

> NB: spams are not constructive and pollute the contributory sites. These contributions are not welcomed and any associated account will be banned immediately. 

Reviewer: I don't think NB is used in English.

```suggestion
Note: spam is not constructive and pollute the contributory sites. These contributions are not welcomed and any associated account will be banned immediately. 
```

The Code of Conduct is intended to apply equally to each member of the community, regardless of its involvement or role in the Project. 
 

> <hr />

Reviewer: ```suggestion
---
```

or

```suggestion
***
```

> This code of conduct is based on the [template](https://github.com/todogroup/opencodeofconduct/blob/gh-pages/codeofconduct_redo.md) established by the [TODO Group](http://todogroup.org/) and used by numerous other large communities (e.g., [Yahoo](https://yahoo.github.io/codeofconduct), [GitHub](http://todogroup.org/opencodeofconduct/#opensource@github.com)), and is influenced by other large online communities like [Stackoverflow](https://stackoverflow.com/conduct). Like the template, this text is shared under the [CC-BY-4.0 license](https://creativecommons.org/licenses/by/4.0/).

Reviewer: Stackoveflow -> Stack Overflow

```suggestion
This code of conduct is based on the [template](https://github.com/todogroup/opencodeofconduct/blob/gh-pages/codeofconduct_redo.md) established by the [TODO Group](http://todogroup.org/) and used by numerous other large communities (e.g., [Yahoo](https://yahoo.github.io/codeofconduct), [GitHub](http://todogroup.org/opencodeofconduct/#opensource@github.com)), and is influenced by other large online communities like [Stack Overflow](https://stackoverflow.com/conduct). Like the template, this text is shared under the [CC-BY-4.0 license](https://creativecommons.org/licenses/by/4.0/).
```
